packages/@ourworldindata/grapher/src/slopeCharts/SlopeChart.tsx

```diff
@@ -1004,26 +1043,63 @@ class LabelledSlopes
 
             this.mouseFrame = requestAnimationFrame(() => {
                 if (this.props.bounds.contains(mouse)) {
-                    const distToSlope = new Map<SlopeProps, number>()
+                    if (this.slopeData.length === 0) return
+
+                    const { x1: startX, x2: endX } = this.slopeData[0]
+
+                    // whether the mouse is over the chart area,
+                    // the left label area, or the right label area
+                    const mousePosition =
+                        mouse.x < startX
+                            ? "left"
+                            : mouse.x > endX
+                            ? "right"
+                            : "chart"
+
+                    const distToSlopeOrLabel = new Map<SlopeProps, number>()
                     for (const s of this.slopeData) {
-                        const dist =
-                            Math.abs(
-                                (s.y2 - s.y1) * mouse.x -
-                                    (s.x2 - s.x1) * mouse.y +
-                                    s.x2 * s.y1 -
-                                    s.y2 * s.x1
-                            ) /
-                            Math.sqrt((s.y2 - s.y1) ** 2 + (s.x2 - s.x1) ** 2)
-                        distToSlope.set(s, dist)
+                        // start and end point of a line
+                        let p1: PointVector
+                        let p2: PointVector
+
+                        if (mousePosition === "chart") {
+                            // points define the slope line
+                            p1 = new PointVector(s.x1, s.y1)
+                            p2 = new PointVector(s.x2, s.y2)
+                        } else if (mousePosition === "left") {
+                            const labelBox = s.leftLabelBounds.toProps()
+                            // points define a "strike-through" line that stretches from
+                            // the left side of the left label to the start point of the slopes
+                            const y = labelBox.y + labelBox.height / 2
+                            p1 = new PointVector(labelBox.x, y)
+                            p2 = new PointVector(startX, y)
+                        } else {
+                            const labelBox = s.rightLabelBounds.toProps()
+                            // points define a "strike-through" line that stretches from
+                            // the end point of the slopes to the right side of the right label
+                            const y = labelBox.y + labelBox.height / 2
+                            p1 = new PointVector(endX, y)
+                            p2 = new PointVector(labelBox.x + labelBox.width, y)
+                        }
+
+                        // calculate the distance to the slope or label
+                        const dist = PointVector.distanceFromPointToLineSegment(
```

Comment:
I'm almost inclined to keep using `distanceFromPointToLineSegmentSq` (emphasis on `Sq`) here, since it's gonna be _a bit_ more performant by not running `Math.sqrt` hundreds of times, and we don't care about the actual distances, just the smallest one (and they are all positive).

Let me know what you think.

If we were to change this, then we would also need to change the tolerance below accordingly, i.e. to `10**2` and `20**2`, respectively.

Reply:
yes, that sounds reasonable. I switched to the non-squared version because I wanted an easy-to-think-about tolerance value but didn't think about perf. let's use the squared version!